src/Interpreters/InterpreterCreateQuery.cpp

```diff
@@ -720,7 +720,7 @@ InterpreterCreateQuery::TableProperties InterpreterCreateQuery::getTableProperti
     }
     else if (create.select)
     {
-        Block as_select_sample = InterpreterSelectWithUnionQuery::getSampleBlock(create.select->clone(), getContext());
+        Block as_select_sample = InterpreterSelectWithUnionQuery::getSampleBlock(create.select->clone(), getContext(), false /* is_subquery */, create.isParameterizedView());
```

Comment:
This seems very bad that such generic method `InterpreterSelectWithUnionQuery::getSampleBlock` now has an argument `is_parametrized_view` 😞. Hmm, but again my last suggestion -- we only need to know this information (as I understand and remember) at the lowest level when checking some ast that it is a `ASTQueryParameter` and we do not need to resolve it, so we can add a method `allow_non_resolved_parameter` into `ASTQueryParameter` and it will solve the issue? (Or if it is needed on a higher level - this flag can be put into `ASTSelectQuery`, but I think it should be possible to do the as I suggested firstly)

Reply:
It would be great if we had something like `interpreter_options`, probably good to take as Improvement PR. But yes, till then we would have to pass these parameters separately as we already have `is_subquery`, actually this is used similar to 'is_create_parameterized_view'. 
We need this information in lowest level, in where level & select level. About keeping the flag in ASTQueryParameters, as mentioned in above comment, it would create the same crutch.